fakebook/src/app/component/profile-view/profile-view.component.ts

```diff
@@ -10,27 +14,75 @@ import { ProfileService } from '../../service/profile.service';
   styleUrls: ['./profile-view.component.css']
 })
 export class ProfileViewComponent implements OnInit {
-  user: User | null = null;
+  user: User | undefined;
+  selfUser!: User;
+  // Person logged in email
+  currentUserEmail = '';
+  // This should be the email of the person whose profile you click on
+  profileUserEmail = '';
+  posts: Post[] = [];
+  followStatus = false;
+  selfProfileCheck = false;
 
-  constructor(
-    private route: ActivatedRoute, // getting the id # in route
-    private profileService: ProfileService
-  ) {}
+  constructor(private oktaAuth: OktaAuthService,
+              private profileService: ProfileService,
+              private route: ActivatedRoute,
+              private followService: FollowService,
+              private postService: PostService) { }
+
+  async ngOnInit(): Promise<void> {
+    // Get Current User Email
+    const userClaims = await this.oktaAuth.getUser();
+    if (userClaims) {
+      this.currentUserEmail = userClaims.email ?? '';
+    }
 
-  ngOnInit(): void {
     this.getUser();
   }
-  getUser(): void {
-    let tempId = ''; // the only way i could declare a variable that may accept a null value in thefuture
-    // get the id number from the route
-    if (this.route.snapshot.paramMap.get('id') != null) {
-      tempId += this.route.snapshot.paramMap.get('id');
 
-      const id = tempId;
+  async getUser(): Promise<void> {
+    let tempEmail = '';
```

Comment:
again, this temp variable is silly. just assign the result of `get()` to a variable and then null-check the variable.

Reply:
done